lms/djangoapps/instructor/views/api.py

```diff
@@ -133,29 +133,32 @@
 
 TASK_SUBMISSION_OK = 'created'
 
+SUCCESS_MESSAGE_TEMPLATE = "The {} report is being created. " \
+                           "To view the status of the report, see Pending Tasks below."
+
 
 def common_exceptions_400(func):
     """
     Catches common exceptions and renders matching 400 errors.
     (decorator without arguments)
     """
+
     def wrapped(request, *args, **kwargs):  # pylint: disable=missing-docstring
         use_json = (request.is_ajax() or
                     request.META.get("HTTP_ACCEPT", "").startswith("application/json"))
         try:
             return func(request, *args, **kwargs)
         except User.DoesNotExist:
-            message = _("User does not exist.")
-            if use_json:
-                return JsonResponse({"error": message}, 400)
-            else:
-                return HttpResponseBadRequest(message)
-        except AlreadyRunningError:
-            message = _("Task is already running.")
-            if use_json:
-                return JsonResponse({"error": message}, 400)
-            else:
-                return HttpResponseBadRequest(message)
+            message = 'User does not exist.'
+        except (AlreadyRunningError, QueueConnectionError) as err:
```

Comment:
Yes, code is refactored to make already running error message more generic. We now use `The {report_type} report is being created. To view the status of the report, see Pending Tasks below. You will be able to download the report when it is complete.` for all types of reports. 
Is that makes sense?